### PR TITLE
8280001: Serial: Add documentation to heap memory layout

### PR DESCRIPTION
**Pull request by albertnetymk**
Comments-only change of adding some documentation.

Test: build

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280001](https://bugs.openjdk.java.net/browse/JDK-8280001): Serial: Add documentation to heap memory layout

### Reviewers
 * [Hamlin Li](https://openjdk.java.net/census#mli) (@Hamlin-Li - **Reviewer**)
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7075/head:pull/7075` \
`$ git checkout pull/7075`

Update a local copy of the PR: \
`$ git checkout pull/7075` \
`$ git pull https://git.openjdk.java.net/jdk pull/7075/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7075`

View PR using the GUI difftool: \
`$ git pr show -t 7075`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7075.diff">https://git.openjdk.java.net/jdk/pull/7075.diff</a>

</details>

### CODE REVIEW

**Review by Hamlin-Li — APPROVED** [review]
I like it, thanks!

**Review by kstefanj — APPROVED** [review]
Looks good.

File: src/hotspot/share/gc/serial/serialHeap.hpp [hunk 1]
```diff
@@ -35,6 +35,26 @@ class MemoryPool;
...
+//
+// low                                                                              high
+//
+//                                              +-- generation boundary (fixed after startup)
+//                                              |
+// |<-    young gen (reserved MaxNewSize)     ->|<- old gen (reserved MaxOldSize) ->|
+// +-----------------+--------+--------+--------+---------------+-------------------+
+// |       eden      |  from  |   to   |        |      old      |                   |
+// |                 |  (to)  | (from) |        |               |                   |
+// +-----------------+--------+--------+--------+---------------+-------------------+
```
Comment (tschatzl) [inline src/hotspot/share/gc/serial/serialHeap.hpp:55]: Unfortunately, the young gen contents are shown wrongly: from/to space are of fixed size and adjacent to the generation boundary.
There is no space between to-space and old gen.
Eden, from and to-space are committed separately within their space and never move.

**Review by tschatzl — CHANGES_REQUESTED** [review]

**Review by tschatzl — APPROVED** [review]
Having had another look at the code, lgtm. Sorry for the confusion.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**